dpctl/tensor/libtensor/include/kernels/elementwise_functions/sign.hpp

```diff
@@ -0,0 +1,267 @@
+//=== sign.hpp -   Unary function SIGN                   ------  *-C++-*--/===//
+//
+//                      Data Parallel Control (dpctl)
+//
+// Copyright 2020-2023 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//===---------------------------------------------------------------------===//
+///
+/// \file
+/// This file defines kernels for elementwise evaluation of SIGN(x) function.
+//===---------------------------------------------------------------------===//
+
+#pragma once
+#include <CL/sycl.hpp>
+#include <cmath>
+#include <cstddef>
+#include <cstdint>
+#include <limits>
+#include <type_traits>
+
+#include "kernels/elementwise_functions/common.hpp"
+
+#include "utils/offset_utils.hpp"
+#include "utils/type_dispatch.hpp"
+#include "utils/type_utils.hpp"
+#include <pybind11/pybind11.h>
+
+#include <iostream>
+
+namespace dpctl
+{
+namespace tensor
+{
+namespace kernels
+{
+namespace sign
+{
+
+namespace py = pybind11;
+namespace td_ns = dpctl::tensor::type_dispatch;
+
+using dpctl::tensor::type_utils::is_complex;
+using dpctl::tensor::type_utils::vec_cast;
+
+template <typename argT, typename resT> struct SignFunctor
+{
+    static_assert(std::is_same_v<resT, argT>);
+    using is_constant = typename std::false_type;
+    // constexpr resT constant_value = resT{};
+    using supports_vec = typename std::negation<
+        std::disjunction<is_complex<resT>, is_complex<argT>>>;
+    using supports_sg_loadstore = std::false_type;
+
+    resT operator()(const argT &x)
+    {
+        if constexpr (std::is_integral_v<argT>) {
+            if constexpr (std::is_unsigned_v<argT>) {
+                return resT(0 < x);
+            }
+            else {
+                return sign<argT>(x);
+            }
+        }
+        else {
+            if constexpr (is_complex<argT>::value) {
+                if (x == argT(0)) {
+                    return resT(0);
+                }
+                else {
+                    return (x / std::abs(x));
+                }
+            }
+            else {
+                if (std::isnan(x)) {
+                    return x;
```

Comment:
```suggestion
                    return std::numeric_limits<resT>::quiet_NaN();
```

This would standardize the returned `NaN`.